scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java

```diff
@@ -364,12 +373,22 @@ public synchronized void pendingToRunning(T js) {
 
     public synchronized void pendingToFinished(T js) {
         pendingJobs.remove(js);
+        putInFinished(js);
+    }
+
+    private void putInFinished(T js) {
+        if (finishedJobs.size() >= maxFinishedJobsSize) {
```

Comment:
I wonder if this routine should be launch in initMaxFinishedJobs() as well:
1/ A user configures scheduler.state.max.finished.jobs to 1000, and used it so he reaches this threshold: Your implementation will work as expected, and limits the finishedJobs to 1000.
2/ The user update  scheduler.state.max.finished.jobs to 2000, and used it so he reaches this threshold: Your implementation will work as expected, and limits the finishedJobs to 2000.
3/ The user prefer to revert back this setting and set scheduler.state.max.finished.jobs back to 1000. Your implementation will still remain w/ 2000 finished Jobs, since it only removes one exceeding jobs when another one is set to be finished.
So ... shouldn't it be worthy to have all the finishedJobs exceeding the scheduler.state.max.finished.jobs threshold removed during initMaxFinishedJobs() to prevent that scenario from happening ?

Reply:
this JVM setting is supposed to be defined at JVM initialization, not to be changed at runtime.
This is why the static field is private and initialized once. If the JVM is restarted, the new setting will be taken into account, and the number of finished jobs kept in SchedulerStateImpl will adapt accordingly. 
The scheduler actually loads 0 finished jobs from the database, the SchedulerStateImpl structure is populated only by jobs submitted after the start.